File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/MotorTesting.java
package org.firstinspires.ftc.teamcode;

import com.qualcomm.robotcore.eventloop.opmode.LinearOpMode;
import com.qualcomm.robotcore.eventloop.opmode.TeleOp;
import com.qualcomm.robotcore.hardware.DcMotorSimple;
import com.qualcomm.robotcore.hardware.Servo;
import com.qualcomm.robotcore.hardware.DcMotor;

// Motor and arm tester

@TeleOp(name="Motor Tester", group="Dev")
public class MotorTesting extends LinearOpMode {

    // Declare members
    private FourWheelRobot robot;

    private DcMotor armLeft;
    private DcMotor armRight;

    private Servo clawLeft;
    private Servo clawRight;

    @Override
    public void runOpMode() {
        robot = new FourWheelRobot(hardwareMap);
        robot.reset();

        armLeft = RobotUtil.getDcMotor(
            hardwareMap,
            "armLeft",
            DcMotorSimple.Direction.REVERSE,
            DcMotor.ZeroPowerBehavior.FLOAT,
            DcMotor.RunMode.RUN_USING_ENCODER
        );
        armRight = RobotUtil.getDcMotor(
            hardwareMap,
            "armRight",
            DcMotorSimple.Direction.FORWARD,
            DcMotor.ZeroPowerBehavior.FLOAT,
            DcMotor.RunMode.RUN_USING_ENCODER
        );

        clawLeft = RobotUtil.getServo(
                hardwareMap,
                "clawLeft",
                Servo.Direction.REVERSE
        );
        clawRight = RobotUtil.getServo(
                hardwareMap,
                "clawRight",
                Servo.Direction.FORWARD
        );

        telemetry.addData("Status", "Initialized");
        telemetry.update();

        // Wait for the game to start (driver presses PLAY)
        waitForStart();

        while (opModeIsActive()) {
            if (gamepad1.a) {
                clawLeft.setPosition(0.0);
                clawRight.setPosition(0.0);
            }
            else if (gamepad1.b) {
                clawLeft.setPosition(1.0);
                clawRight.setPosition(1.0);
            }

            telemetry.addData("arm left", armLeft.getCurrentPosition());
            telemetry.addData("arm right", armRight.getCurrentPosition());
            telemetry.addData("left front wheel", robot.leftFront.getCurrentPosition());
            telemetry.addData("left trigger", gamepad1.left_trigger);

            telemetry.addData("left claw", clawLeft.getPosition());
            telemetry.addData("right claw", clawRight.getPosition());
            telemetry.update();
        }
    }

    private double armPosition;
    private final double maxArmPos = 100.0;
    private final double minArmPos = 0.0;

    private void initArm() {
        setArmPosition(minArmPos);
        armLeft.setMode(DcMotor.RunMode.RUN_TO_POSITION);
        armRight.setMode(DcMotor.RunMode.RUN_TO_POSITION);
        armLeft.setPower(0.5);
        armRight.setPower(0.5);
    }
    private void setArmPosition(double ticks) {
        // Clamp arm position between min and max.
        // ticks = Math.max(minArmPos, Math.min(ticks, maxArmPos));

        int output = (int) Math.round(ticks);
        armLeft.setTargetPosition(output);
        armRight.setTargetPosition(output);
        armPosition = ticks;
    }
    private void shiftArmPosition(double dTicks) {
        setArmPosition(armPosition + dTicks);
    }



}

File: TeamCode/src/main/kotlin/org/firstinspires/ftc/teamcode/machines/MotorTesting.kt
package org.firstinspires.ftc.teamcode.machines

import com.qualcomm.robotcore.eventloop.opmode.TeleOp
import com.qualcomm.robotcore.eventloop.opmode.LinearOpMode

import org.firstinspires.ftc.teamcode.common.LateInitConstProperty
import org.firstinspires.ftc.teamcode.common.leftTriggerPressed
import org.firstinspires.ftc.teamcode.common.rightTriggerPressed

// Motor and arm tester
@TeleOp(name = "Motor Tester", group = "Dev")
class MotorTesting : LinearOpMode() {
    // Declare members
    private var robot: MadMachinesRobot by LateInitConstProperty()

    override fun runOpMode() {
        robot = MadMachinesRobot(hardwareMap)
        robot.reset()

        telemetry.addData("Status", "Initialized")
        telemetry.update()

        // Wait for the game to start (driver presses PLAY)
        waitForStart()
        while (opModeIsActive()) {
//            when {
//                gamepad1.a -> {
//                    robot.clawLeft.position -= 0.001
//                    robot.clawRight.position -= 0.001
//                }
//                gamepad1.b -> {
//                    robot.clawLeft.position += 0.001
//                    robot.clawRight.position += 0.001
//                }
//            }

            for (lol in arrayOf(
                Pair(gamepad2.left_bumper, robot.leftFront),
                Pair(gamepad2.right_bumper, robot.rightFront),
                Pair(gamepad2.leftTriggerPressed, robot.leftRear),
                Pair(gamepad2.rightTriggerPressed, robot.rightRear),
            )) {
                lol.second.power =
                    if (lol.first) 0.5 else 0.0
            }

            robot.clawLeft.position += when {
                gamepad1.leftTriggerPressed -> -0.001
                gamepad1.left_bumper -> 0.001
                else -> 0.0
            }
            robot.clawRight.position += when {
                gamepad1.rightTriggerPressed -> -0.001
                gamepad1.right_bumper -> 0.001
                else -> 0.0
            }
            when {
                gamepad1.dpad_up -> {
                    for (motor in arrayOf(robot.armLeft, robot.armRight))
                        motor.targetPosition += 1
                }
                gamepad1.dpad_down -> {
                    for (motor in arrayOf(robot.armLeft, robot.armRight))
                        motor.targetPosition -= 1
                }
            }
            val thingsToPrint: Array<Pair<String, Any?>> = arrayOf(
                "arm left" to robot.armLeft.currentPosition,
                "arm right" to robot.armRight.currentPosition,
                "left front wheel" to robot.leftFront.currentPosition,
                "left trigger" to gamepad1.left_trigger,
                "left claw" to robot.clawLeft.position,
                "right claw" to robot.clawRight.position,
            )
            for ((caption, content) in thingsToPrint)
                telemetry.addData(caption, content)
            telemetry.update()
        }
    }
}
